src/main/scala/is/hail/annotations/MemoryBlock.scala

```diff
@@ -522,15 +522,13 @@ class RegionValueBuilder(var region: MemoryBuffer) {
 
   def start(newRoot: Type) {
     assert(inactive)
-
     root = newRoot.fundamentalType
+  }
+
+  def allocateRoot() {
     root match {
       case t: TArray =>
-        region.align(t.contentsAlignment)
```

Comment:
These (cases TArray and TBinary) got removed because the alignment happens already in startArray and addBinary, right? I don't see `start` getting anywhere in these cases, but perhaps I'm missing it? are there tests for arrays + more basic types as region values?

Reply:
You're quite right, good catch.

 > These (cases TArray and TBinary) got removed because the alignment happens already in startArray and addBinary, right?

Correct, it was duplicate (I was never happy  about that).  But I wasn't aligning in fixup{Array, Binary}, fixed.  x86 supports unaligned accesses, so that won't cause a failure.  We might consider alignment checks in MemoryBuffer, although we intentionally use unaligned accesses in the en/decoder.

 > I don't see start getting anywhere in these cases, but perhaps I'm missing it?

Nope, that was just missing.  Should be fixed.  I'm a little confused how the tests passed.  I need to think about the tests more.

 > are there tests for arrays + more basic types as region values?

Yes, UnsafeSuite.testRegionValue tests examples of all types including arrays and scalars (it uses `Type.genArb` which generates arbitrary types).